### ISSUE TITLE
Broken avatar on GHE (again)

### ISSUE THREAD

**Issue by LunaticMuch** [gitify-app/gitify#1224]
### 🔍 Is there already an issue for your problem?

- [X] I have checked older issues, open and closed

### 📝 Description

In a GHE situation, avatars for users are broken. 
This seems similar to #564 but in fact is not. 

Looking at the inspector tab, the `src` for the image is correctly populated and points a valid image if I open it from my browser.
Incidentally, I have an amount of warnings as 

> reading cookie in cross-site context will be blocked in future Chrome versions

Whether that's the real problem or not...I am yet to understand. I cannot find any other error.

### 🪜 Steps To Reproduce

1. Use in a GHE context
2. GHE is 3.11

### Gitify Version

5.7.0

### Operating System

macOS

### GitHub Account

GitHub Enterprise

### 📸 Screenshots

![Screenshot 2024-06-12 at 10 39 10](https://github.com/gitify-app/gitify/assets/1659414/e9e032d7-5d52-4465-9a85-fe6ca0efe78d)

**Comment by setchy** [comment]
thanks for reporting @LunaticMuch

so to confirm, the avatar url returns a valid image, but renders as broken  🤔

**Comment by LunaticMuch** [comment]
> so to confirm, the avatar url returns a valid image, but renders as broken 🤔

Yes. I checked the code, the console, but I can get my head around it. I only suspected the cookie, but also I cannot see the "relationship"

**Comment by setchy** [comment]
Puzzling indeed... together I'm sure we'll get to the bottom of it 🤝 

Any chance you are able to run gitify from the source code to see if that has the same issue?  
```
git clone
pnpm i
pnpm watch
pnpm start
```

**Comment by bmulholland** [comment]
I also wonder whether there's a network request for the image at all? Maybe there's a security policy that prevents the loading, for instance. Would need @LunaticMuch to check that, I don't have GHE to try to repro

**Comment by setchy** [comment]
> @setchy and @bmulholland I can definitely help... give me a bit, feels more a debug I need to squeeze after work 😄

absolutely!  thank you again @LunaticMuch 🙇

**Comment by LunaticMuch** [comment]
First debug session, following your guidance @setchy 

1. Console is empty. There's no reported error or warning. Actually nothing at all.
2. I do not see any call for the image tracing the network... this quite interesting, but it feels weird the security policy. If the connection is not started, this should not be related to a policy. The policy might block the response, not opening a socket for the call

**Comment by setchy** [comment]
Looks like you have `Detailed Notifications` enabled.  In that case, you'll find the logic within `src/utils/subject.ts` that enriches notifications with the most recent `SubjectUser` (logic varies by notification type ie: Issues vs PullRequests vs Discussions, etc)

**Comment by LunaticMuch** [comment]
I need to dig into that, but from my initial testing, it's all fine. 
Actually, the repo icon suffers of the same problem, but the default icon for the runners does not 🤔 

![Screenshot 2024-06-13 at 12 44 03](https://github.com/gitify-app/gitify/assets/1659414/2bc08a49-8c8f-47e0-9f7e-e3999469da13)

**Comment by setchy** [comment]
The default icons used for workflows are from the primer/octicons package

**Comment by setchy** [comment]
Perhaps a fix lies in setting the crossorigin attribute on the img tags...

https://www.geeksforgeeks.org/how-to-allow-cross-origin-use-of-images-and-canvas/

@LunaticMuch  - when you said these urls load in a new tab, was that in an incognito/private or a regular instance?

**Comment by setchy** [comment]
@LunaticMuch - just wanted to confirm if you're still experiencing this issue?

**Comment by setchy** [comment]
Closing as stale. Happy to reopen if others report more details to help triage

**Comment by mislavperi** [comment]
I can confirm the issue still occurs as of today

**Comment by setchy** [comment]
@mislavperi - are you able to share further details please, such as Gitify version, GitHub account type(s), OS type.  Any log screenshots would also be useful (via Developer Tools network console or app log file)

**Comment by mislavperi** [comment]
@setchy Yes I'll get you those today

**Comment by setchy** [comment]
Perfect!

I've also been wondering if these broken avatars also occur on this feature branch which uses the official GitHub Primer avatar components 

https://github.com/gitify-app/gitify/pull/1589

**Comment by mislavperi** [comment]
@setchy 
Account type enterprise account, we run on GHE 3.13.3
Running MacOS Sonoma 14.3
Gitlify version is v5.16.1
It also seems to be stuck on high number of notifications (I can't clear them at all) and new ones never come
![Screenshot 2024-10-29 at 09 13 58](https://gith
![Screenshot 2024-10-29 at 09 13 11](https://github.com/user-attachments/assets/6e088c42-607b-4a83-86d2-f3451c12c4ef)
ub.com/user-attachments/assets/30787a51-80c3-441f-b491-6c5a733918e8)
Here are the inspect element network tab and console errors I get

**Comment by setchy** [comment]
HTTP 302 hey... So is there a suggested new location? 

If you get the chance to test Gitify locally using that branch I shared, I'd love to hear if this is natively resolved within the GitHub component library

**Comment by mislavperi** [comment]
![image](https://github.com/user-attachments/assets/40c2f0d2-99f3-4109-b41e-8898de6d8dd0)
@setchy 
Still the same. I'm also struggling to open the inspect element for some reason

**Comment by setchy** [comment]
Thanks for testing the branch @mislavperi. 🙏

Back to the drawing board then 😅. 

I'm curious if this happens for ALL GHE users or a smaller cohort 🤔

**Comment by mislavperi** [comment]
Let me know if you need any help @setchy

**Comment by setchy** [comment]
We'd love a hand @mislavperi since none of the maintainers are able to recreate this bug and find an effective solution.  If you are able to spend some time digging deeper that would be greatly appreciated. 

---
As an aside, I've been thinking about interim "hacks" to address the broken avatar images experience which is not ideal...  perhaps we should use the default GitHub Mark Icon avatar (either for ALL GHE accounts, or, through a user setting)... at least until we find the root of the issue

**Comment by setchy** [comment]
@pddg - as a GHE user, do you also have this broken avatar issue?

**Comment by pddg** [comment]
The avatar image was corrupted when a problem occurred connecting to the GHE Server.
Even if the connection is restored, it image will not be restored immediately. However, it will be displayed normally at some point.

Open/Close window, reload manually does not help. Open settings and return to the notifications list will recover the avatar image.
I hadn't really thought about it before, so there may be other cases where it is restored.

**Comment by setchy** [comment]
Ah ha, that's helpful insight @pddg.

**Comment by mislavperi** [comment]
@setchy In my case, it refuses to connect after initial configuration was defined, I'll spin up the instance locally and see why it might be happening

**Comment by mislavperi** [comment]
Sorry for my long absence, but I found generally that loading avatars with GHE is generally a big pain
https://github.com/microsoft/vscode-pull-request-github/issues/416
I found other countless threads that showcase the same issue. Do you maybe want to introduce fallback icons in case there is a 302?

**Comment by setchy** [comment]
@mislavperi - is this something you'd be interested in contributing back?

**Comment by mislavperi** [comment]
@setchy gladly, I'll open a feature branch and play around it, see what might be the best way to implement it

**Comment by mislavperi** [comment]
@setchy I believe I found the issue. I investigate the request made for the avatar and came to the conclusion the authorization header is missing
I copied the request, added the authorization header and I got this
![image](https://github.com/user-attachments/assets/446ac9a7-1b84-41a8-9ca1-3b0eda1c191a)

**Comment by mislavperi** [comment]
@setchy another update, turns out it's not the authorization but rather the cookie header that has to be set. I'm not sure that we can get that injected but I will further explore what can be done

**Comment by mislavperi** [comment]
Specifically, difference in cookies in web vs gitify version is big
gitify
![image](https://github.com/user-attachments/assets/efb153a0-6a4a-448c-81ed-e21b701c8248)
web version
![image](https://github.com/user-attachments/assets/de3df4b3-08a8-4653-a199-18060b9b2ca5)
my guess, is that PAT specifically doesn't get those since technically we are not logged, only using PAT. I'm not 100% sure how to work around this, I'll do a bit more research

**Comment by mislavperi** [comment]
@setchy do you think the app would benefit from logging into enterprise github instances directly?
